### ISSUE TITLE
Remove Node.from_parent/.from_config indirection

### ISSUE THREAD

**Issue by blueyed** [pytest-dev/pytest#6645]
This is meant to go forward with https://github.com/pytest-dev/pytest/pull/6571#issuecomment-580761091 / unblock the next release.

Ref: https://github.com/pytest-dev/pytest/pull/5975#discussion_r339523282

Proof-of-concept: https://github.com/blueyed/pytest/pull/143

TODO:

- [ ] this is based on the first addition of `from_parent` (#5975), but I expect 8ba0b7b (#6387) can be done in a similar way
- [ ] only allow particular classes (hierarchy)?  (i.e. `Module(parent=Item(…))` makes no sense)
- [ ] https://github.com/pytest-dev/pytest/pull/6293 - but not clear how / if that is planned to be done like that still (should it be rebased/adjusted first?)

**Comment by RonnyPfannschmidt** [comment]
Not sure what you are doing here, this is failing the big picture at first glance and that won't do

**Comment by blueyed** [comment]
Ok, let's maybe move on then?
=> https://github.com/pytest-dev/pytest/pull/6571#issuecomment-582111483
